lib/dal/tests/integration_test/component.rs

```diff
@@ -906,3 +910,81 @@ async fn undoing_deletion_updates_inputs(ctx: &mut DalContext) {
     assert!(!units_json_string.contains("docker.io/library/oysters in my pocket\\n"));
     assert!(units_json_string.contains("docker.io/library/were saving for lunch\\n"));
 }
+
+#[test]
+async fn paste_component(ctx: &mut DalContext) {
+    let pirate_name = "Long John Silver";
+    let parrot_name = "Captain Flint";
+    let pirate_component = create_component_for_schema_name(ctx, "pirate", pirate_name).await;
+
+    let parrots_path = &["root", "domain", "parrot_names"];
+
+    let pet_shop_component = create_component_for_schema_name(ctx, "pet_shop", "Petopia").await;
+
+    // set value on source component
+    {
+        let pet_shop_parrot_av_id = pet_shop_component
+            .attribute_values_for_prop(ctx, parrots_path)
+            .await
+            .expect("find value ids for prop parrot_names")
+            .pop()
+            .expect("there should only be one value id");
+
+        AttributeValue::insert(ctx, pet_shop_parrot_av_id, Some(parrot_name.into()), None)
+            .await
+            .expect("insert value in pet_shop parrot_names array");
+    }
+
+    connect_components_with_socket_names(
+        ctx,
+        pet_shop_component.id(),
+        "parrot_names",
+        pirate_component.id(),
+        "parrot_names",
+    )
+    .await;
+
+    commit_and_update_snapshot(ctx).await;
+
+    let pasted_pirate_component = pirate_component
+        .copy_paste(ctx, (20., 20.))
+        .await
+        .expect("unable to paste component");
+
+    commit_and_update_snapshot(ctx).await;
+
+    let mut view = pasted_pirate_component
+        .materialized_view(ctx)
+        .await
+        .expect("unable to get materialized view of component")
+        .expect("no view found");
+
+    // Non deterministic
+    *view
+        .pointer_mut("/resource/last_synced")
+        .expect("no resource last_synced found") = serde_json::Value::Null;
```

Comment:
The whole test is non deterministic or just this value?

As an aside, we probably want to restore the old `ComponentViewProperties` builder (or something like it) that could drop individual fields, if you remember it. It was a way to remove timestamp fields and non-deterministic fields from the `ComponentView`. We probably want something similar for the materialized view.

Reply:
last_synced uses the clock, so it may not be deterministic

It would be cool doing that with materialized view yeah

Reply:
Removed the entire resource